### PR TITLE
fix(azure_ad): make redirect and graph_url optional parameters and update docs

### PR DESCRIPTION
**Pull request by aditya-radhakrishnan**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [x] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/source_docs/azure-ad.md [hunk 1]
```diff
@@ -116,11 +121,11 @@ Note that a `.` is used to denote nested fields in the YAML configuration block.
...
 | `token_url`                            | string          | ✅        |                       | The token URL that acquires a token from Azure AD for authorizing requests.  This source will only work with v1.0 endpoint.                                                                                                                                |
-| `graph_url`                            | string          | ✅        |                       | [Microsoft Graph API endpoint](https://docs.microsoft.com/en-us/graph/use-the-api)                                                                                                                                                                         |
+| `redirect`                             | string          |          |                       | Redirect URI.  Found in your app registration on Azure AD Portal                                                                                                                                                                                           |
```
Comment (jjoyce0510) [inline metadata-ingestion/source_docs/azure-ad.md:127]: minor: let's put in the comment what we default to if not provided.

File: metadata-ingestion/src/datahub/ingestion/source/identity/azure_ad.py [hunk 2]
```diff
@@ -228,7 +230,9 @@ def _add_group_members_to_group_membership(
                     )
                 else:
                     # Unless told otherwise, we only care about users and groups.  Silently skip other object types.
-                    pass
+                    logger.warning(
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/identity/azure_ad.py:233]: thank you!

**Review by jjoyce0510 — COMMENTED** [review]
Minor comments, otherwise LGTM

File: metadata-ingestion/source_docs/azure-ad.md (hunk 1 above)
Comment (aditya-radhakrishnan) [reply metadata-ingestion/source_docs/azure-ad.md]: Done!

**Review by jjoyce0510 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**